blog/_posts/2018-03-29-missing.md

```diff
@@ -0,0 +1,414 @@
+  (similar to `DataArray`).
+
+For all these reasons, `Nullable`
+[no longer exists](https://github.com/JuliaLang/julia/pull/23642)
+in Julia 0.7. Several replacements are provided, depending on the use case:
+
+- Data which can contain statistical missing values should be represented as
+  `Union{Missing,T}`, i.e. either a value of type `T` or the `missing` object.
+
+- Situations where there may be a value of type `T` or no value should use
+  `Union{Nothing,T}` (equivalent to `Union{Void,T}` on Julia 0.6). As a special case,
+  if `nothing` is a possible value (i.e. `Nothing <: T`), `Union{Nothing,Some{T}}`
+  should be used instead. This pattern is used by e.g. `findfirst` and `tryparse`.
+
+This blog post covers the first case, and should hopefully make it clear why
+it is useful to distinguish `missing` and `nothing`. To give a first insight, let
+us note that the main difference between these two objects is that `missing`
+generally propagates when passed to standard mathematical operators and functions,
+while `nothing` does not implement any specific method and therefore generally
+gives a `MethodError`.
+
+The rest of the post illustrates the expression "first class support" used
+in the title by presenting three properties of the Julia 0.7 implementation of
+statistical missing values:
+
+1. The `missing` object can be used in combination with any type, be it defined in
+   Base, in a package or in user code.
+
+2. Standard Julia code working with missing values is efficient, without special tricks.
+
+3. Missing values are safe by default: when passed to most functions, they either
+   propagate or throw an error.
+
+Finally, current limitations and future improvements are discussed.
+
+## A generic representation
+
+One of Julia's strengths is that user-defined types are as powerful and fast as built-in
+types. To fully take advantage of this, missing values had to support not only standard
+types like `Int`, `Float64` and `String`, but also any custom type. For this reason,
+Julia cannot use the so-called *sentinel* approach like R and Pandas to represent
+missingness, that is reserving special values within a type's domain. For example,
+R represents missing values in integer and boolean vectors using the smallest
+representable 32-bit integer (`-2,147,483,648`), and missing values in floating point
+vectors using a specific `NaN` payload (`1954`, which rumour says refers to Ross Ihaka's
+year of birth). Pandas only supports missing values in floating point vectors,
+and conflates them with `NaN` values.
+
+In order to provide a consistent representation of missing values which can be combined
+with any type, Julia 0.7 will use `missing`, an object with no fields which is the only
+instance of the the `Missing` singleton type. This is a normal Julia type with a few
+peculiarities which are detailed below. Values which can be either of type `T` or missing
```

Comment:
Not a fan of the word `peculiarities` here, which carries just a slight negative connotation; like `missing` had to be special-cased in the core language or something (which isn't true; just small unions have been optimized).

Reply:
I was thinking about `promote_typejoin`, but I realize I haven't mentioned it. That's quite technical, so maybe a footnote will be enough. Anyway I can remove "peculiarities".

Reply:
You have a “the the” here on line 136.